### PR TITLE
CI: re-introduce sharding in PHPUnit jobs

### PR DESCRIPTION
**Pull request by kalessil**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

In this PR, we re-iterate on https://github.com/woocommerce/woocommerce/pull/39302, and introduce 2 shards: for "legacy" and main test cases.

We also re-group PHP-jobs again:
- WordPress L-2 jobs are dropped (based on team chat and `Requires at least` plugin header)
- WordPress nightly builds are kept in PR as they will be switched to RC instead later this year

### Testing

- CI checks shall pass

### CODE REVIEW

File: plugins/woocommerce/tests/legacy/unit-tests/rest-api/Helpers/ProductHelper.php [hunk 1]
```diff
@@ -24,10 +24,11 @@ class ProductHelper {
 
 	/**
 	 * Counter to insert unique SKU for concurrent tests.
+	 * The starting value ensures no conflicts between existing generators.
 	 *
 	 * @var int $sku_counter
 	 */
-	private static $sku_counter = 0;
+	private static $sku_counter = 100000;
```
Comment (kalessil) [inline plugins/woocommerce/tests/legacy/unit-tests/rest-api/Helpers/ProductHelper.php:31]: Here and other test files: fixes failing tests in main suite surfaced after sharding introduction.

**Comment by adimoldovan** [comment]
I think we can use the `shardingArguments` config instead of creating 2 jobs:

```
"shardingArguments": [
	"--testsuite=wc-phpunit-legacy",
	"--testsuite=wc-phpunit-main"
]
```

**Comment by kalessil** [comment]
Love the proposal! Will update the PR - will be ready for another review round tomorrow morning.

**Comment by kalessil** [comment]
@adimoldovan: The PR has been updated and is ready for the next review round.

**Review by adimoldovan — APPROVED** [review]
LGTM!

**Pull request merged**